src/components/map-marker/map-marker.tsx

```diff
@@ -62,16 +69,38 @@ export class MapMarker implements GxComponent {
    */
   @Event() gxMapMarkerDeleted: EventEmitter;
 
+  private setPopup() {
+    const popupHtml = this.element.querySelector("[class='popupHtml']");
+    if (popupHtml.innerHTML) {
+      const marginProportion = 83 / 100;
+      const maxPopupSize = {
+        height:
+          document.querySelector(".gxMap").clientHeight * marginProportion,
+        width: document.querySelector(".gxMap").clientWidth * marginProportion
+      };
+      this.markerInstance.bindPopup(popupHtml, {
+        keepInView: true,
+        maxHeight: maxPopupSize.height,
+        maxWidth: maxPopupSize.width,
+        minWidth: 100
+      });
+    }
+  }
+
   componentDidLoad() {
-    const halfIconWidth = this.iconWidth / 2;
+    const halfIconSizes = {
+      height: this.iconHeight / 2,
+      width: this.iconWidth / 2
+    };
     const coords = parseCoords(this.coords);
     if (coords !== null) {
```

Comment:
Shouldn't  we emit the `gxMapMarkerDidLoad` inside this `if` block?

Reply:
```gxMapMarkerDidLoad``` must be emitted after process all data and marker structure. ```gxMapMarkerDidLoad``` is successfully emitted at the end of ```componentDidLoad``` method.

Reply:
Sorry, my bad.